libs/features/georoutingv2/georoutingv2.js

```diff
@@ -281,12 +283,19 @@ export default async function loadGeoRouting(
   loadBlock = loadBlockFunc;
   loadStyle = loadStyleFunc;
 
-  const resp = await fetch(`${config.contentRoot ?? ''}/georoutingv2.json`);
+  let resp = await fetch(`${config.contentRoot ?? ''}/georoutingv2.json`);
   if (!resp.ok) {
-    // eslint-disable-next-line import/no-cycle
-    const { default: loadGeoRoutingOld } = await import('../georouting/georouting.js');
-    loadGeoRoutingOld(config, createTag, getMetadata);
-    return;
+    resp = await fetch(`${config.contentRoot ?? ''}/georouting.json`);
+    if (!resp.ok) {
+      resp = await fetch(`${getFederatedContentRoot()}/federal/georouting/georoutingv2.json`);
+      if (!resp.ok) return;
+    } else {
+      const json = await resp.json();
+      // eslint-disable-next-line import/no-cycle
+      const { default: loadGeoRoutingOld } = await import('../georouting/georouting.js');
+      loadGeoRoutingOld(config, createTag, getMetadata, json);
+      return;
+    }
```

Comment:
I don't have the full context of why the old georouting is required, but here a suggestion that would provide a bit more clarity on the load order, avoiding a nested if/else structure:

```js
const urls = [
  `${config.contentRoot ?? ''}/georoutingv2.json`,
  `${config.contentRoot ?? ''}/georouting.json`,
  `${getFederatedContentRoot()}/federal/georouting/georoutingv2.json`
];

for (const url of urls) {
  let resp = await fetch(url);
  if (resp.ok) {
    if (url.includes('georouting.json')) {
      const json = await resp.json();
      // eslint-disable-next-line import/no-cycle
      const { default: loadGeoRoutingOld } = await import('../georouting/georouting.js');
      loadGeoRoutingOld(config, createTag, getMetadata, json);
    }
    return;
  }
}
```

The way I understand it, we want to load georoutingv2, if that's not available the legacy georouting and then the federal georouting as backup?

Reply:
Yes, correct. Agree look cleaner and readable, will update it.